datastore/google/cloud/datastore/key.py

```diff
@@ -380,7 +380,7 @@ def parent(self):
         return self._parent
 
     def __repr__(self):
-        return '<Key%s, project=%s>' % (self.path, self.project)
+        return '<Key%s, project=%s>' % (self._flat_path, self.project)
 
```

Comment:
I didn't do it because it was better, I did it to make it deterministic since `repr(self.path)` depends on `PYTHONHASHSEED` (`path` is a dict).

However, it **is** also better, because a key is really just `(kind, id_or_name, kind, id_or_name, ...)` and the fact that we use a dictionary for each `kind / id_or_name` pair is just an implementation detail.